source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -0,0 +1,100 @@
+#include <fcntl.h>
+#include <io.h>
+#include <sys/stat.h>
+#include <windows.h>
+
+#undef DELETE
+#undef GetMessage
+
+#include "common/common/assert.h"
+#include "common/filesystem/filesystem_impl.h"
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "envoy/common/exception.h"
+
+#include "common/common/fmt.h"
+
+namespace Envoy {
+namespace Filesystem {
+
+FileImplWin32::~FileImplWin32() {
+  if (isOpen()) {
+    const Api::IoCallBoolResult result = close();
+    ASSERT(result.rc_);
```

Comment:
Is rc_ a bool in this case, rather than integer? Making this an explicit comparison rather than implied boolean conversation might help the reader here.

Reply:
Yes - rc_ is a bool. See the definition in the following link:
https://github.com/envoyproxy/envoy/pull/6072/files#diff-c3c5edef651a44eb51816723c183eda1R84